### DISCUSSION TITLE
snapshotter controller redundant?

### DISCUSSION THREAD

**Discussion by fr6nco** [kubernetes-sigs/azuredisk-csi-driver#1414] [Q&A] [answered]
Hi,

i'm doing some tests with the azuredisk and both azurefile csi drivers. I'm looking into snapshots right now. If I set snapshot.enalbed == true CRDs to support volume snapshots are installed by helm. At the same time a csi-snapshot-controller is installed too, however this controller feels a bit redundant, since there is a csi-snapshotter container running in the azure disk controller too.
Looking at logs I see that both controllers will try to execute the snapshot creation / deletion commands.
 
My question is, whats the purpose of the snapshotter-controller if there's already a csi-snapshotter container running in the azuredisk pod?

Thanks.

**Comment by andyzhangx** [comment, marked as answer]
it's not redundant,  both controller and sidecar containers should be installed, you could find details here: https://github.com/kubernetes-csi/external-snapshotter#usage

**Reply by fr6nco** [reply]
All clear now. Thank you :)
